libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/apple/AppleXcodeTasks.kt

```diff
@@ -200,4 +203,32 @@ private val Framework.namePrefix: String
     )
 
 private fun Project.appleFrameworkDir(frameworkSearchDir: File) =
-    buildDir.resolve("xcode-frameworks").resolve(frameworkSearchDir)
\ No newline at end of file
+    buildDir.resolve("xcode-frameworks").resolve(frameworkSearchDir)
+
+/**
+ * macOS frameworks contain symlinks which are resolved/removed by the Gradle [Copy] task.
+ * To preserve these symlinks we are using the `cp` command instead.
+ * See https://youtrack.jetbrains.com/issue/KT-48594.
+ */
+private abstract class FrameworkCopy : DefaultTask() {
+
+    @get:InputFiles
+    var files: FileCollection? = null
```

Comment:
We need little bit more checks :) and since the task is abstract we can avoid nullability
```
@get:PathSensitive(PathSensitivity.ABSOLUTE)
@get:IgnoreEmptyDirectories
@get:InputFiles
@get:SkipWhenEmpty
abstract var files: FileCollection

@get:OutputDirectory
abstract var destDir: File
```